VideoSyncClass/VideoSync.py

```diff
@@ -0,0 +1,283 @@
+        for i,item in enumerate(paths):
+            if not type(item) == str:
+                raise TypeError("Could not set audio paths to the provided path. Item " + str(item) + " at index " + str(i) + " is not a string")
+            else:
+                temp.append(item)
+            if not os.path.exists(os.path.join(self.__input_folder, item)):
+                print("Warning: \'"+ item + "\' could not be found as an item or directory.")
+        self.__audio_paths = temp
+
+    def set_meg_paths(self, paths):
+        """
+        Summary: Sets meg_paths to a list of paths supplied as arguments.
+        """
+        temp = []
+        if type(paths) != list:
+            raise TypeError("Cannot set meg paths to given input. Expected list, recieved " + type(paths) + " Paths supplied must be contained within a list in format [s1, s2, ..., sn].")
+        for i,item in enumerate(paths):
+            if not type(item) == str:
+                raise TypeError("Could not set meg paths to the provided path. Item " + str(item) + " at index " + str(i) + " is not a string")
+            else:
+                temp.append(item)
+            if not os.path.exists(os.path.join(self.__input_folder, item)):
+                print("Warning: \'"+ item + "\' could not be found as an item or directory.")
+        self.__meg_paths = temp
+
+    def set_input_folder(self, path):
+        """
+        Summary: Sets the input folder to the specified path.
+        
+        Arguments:
+            path: The path of the input directory
+        """
+        if type(path) != str:
+            raise TypeError('Path must be provided as a string.')
+        if not os.path.exists(path):
+            print("Warning: "+ path +" could not be found as a file or directory.")
+        self.__input_folder == path
+
+    def set_output_folder(self,path):
+        """
+        Summary: Sets the output folder to the specified path. This is the location where outputs will be stored. Will create this
+                 directory if it does not already exist.
+        
+        Arguments:
+            path: The path of the desired output directory
+        """
+        if type(path) != str:
+            raise TypeError('Path must be provided as a string.')
+        if not os.path.exists(path):
+            print("Warning: "+ path +" could not be found as a file or directory.")
+        self.__output_folder == path
+
+    def parse_files_from_folder(self):
+        """
+        Summary: If an input folder has been declared, read through this folder and set paths to each of the videos, audio, and 
+        """
+        files = os.listdir(self.__input_folder)
+        audio_paths = []
+        video_paths = []
+        meg_paths = []
+        for file in files:
+            ext = os.path.splitext(file)[1]
```

Comment:
consider using the `pathlib.Path` interface rather than `os.path`.  It's generally considered "the modern replacement for os.path" so new code should probably use it unless there's a strong need to stick with os.path